qcodes/instrument_drivers/rigol/DG1062.py

```diff
@@ -290,7 +304,32 @@ def _set_waveform_params(self, **params_dict: Dict) ->None:
         string += ",".join(
             ["{:7e}".format(params_dict[param]) for param in param_names])
         self.parent.write_raw(string)
+        
+    def _get_duty_cycle(self):
+        """
+        Simple function to extract the duty cycle after checking waveform
+        """
+        self.waveform()
+        wf = self.waveform.get_latest()
```

Comment:
in case it's necessary for `_get_duty_cycle` function to operate on the real actual latest waveform of the instrument, then `wf = self.waveform.get_latest()` is redundant because `wf = self.waveform()` does it already.

in case it's necessary for `_get_duty_cycle` function to avoid additional communication with the instrument, then `self.waveform()` should be removed because it does perform the communication; while `wf = self.waveform.get_latest()` only does the job of returning the "latest value of the parameter that is known to the driver (yes, the state of the instrument might have changed since the value of the parameter has been requested due to, for example, manually clicking buttons of the instrument)".

So, depending on the intention, please, get rid of either of the lines.